src/Deserialization/Deserializer.cs

```diff
@@ -1,26 +1,34 @@
-﻿using System;
-using System.IO;
-using System.Text.Json;
-using System.Threading.Tasks;
+﻿using LLVMSharp.Interop;
 using Symbolica.Abstraction;
-using Symbolica.Deserialization.DTOs;
+using Symbolica.Expression;
 
 namespace Symbolica.Deserialization
 {
     public static class Deserializer
     {
-        internal static readonly JsonSerializerOptions Options = new()
+        public static IModule DeserializeModule(byte[] bytes)
         {
-            PropertyNameCaseInsensitive = true,
-            MaxDepth = int.MaxValue
-        };
+            return DeserializeModule(new UnsafeContext(), bytes);
+        }
 
-        public static async Task<IModule> DeserializeModule(Stream stream)
+        private static IModule DeserializeModule(IUnsafeContext unsafeContext, byte[] bytes)
```

Comment:
Minor, but is there much point to this `private` method, given that it seems to only be called from the `public` one above as in a one-liner. I see you're hiding away the creation of the `UnsafeContext` which seems good, but could it not be combined with the rest of this method's implementation?

Reply:
Haha, I did this for a reason that will really annoy you. If I construct the `UnsafeContext` in the same method as the calls to `GetMemoryBuffer`, `GetTargetData`, `GetPointerSize` then these methods are never called via the interface, only directly on the known class. So then Rider tells me to just remove them from the interface, and since they don't use any instance data it then tells me to make them static methods on the class. I could do that but instead I just passed it into a method that only knows about the interface.

Reply:
Lol. OOP overkill.